prefect_gcp/worker.py

```diff
@@ -0,0 +1,642 @@
+        description=(
+            "The length of time that Prefect will wait for a Cloud Run Job to complete "
+            "before raising an exception."
+        ),
+    )
+    keep_job: Optional[bool] = Field(
+        default=False,
+        title="Keep Job After Completion",
+        description="Keep the completed Cloud Run Job on Google Cloud Platform.",
+    )
+
+    @property
+    def project(self) -> str:
+        """property for accessing the project from the credentials."""
+        return self.credentials.project
+
+    @property
+    def job_name(self) -> str:
+        """property for accessing the name from the job metadata."""
+        return self.job_body["metadata"]["name"]
+
+    def prepare_for_flow_run(
+        self,
+        flow_run: "FlowRun",
+        deployment: Optional["DeploymentResponse"] = None,
+        flow: Optional["Flow"] = None,
+    ):
+        """
+        Prepares the job configuration for a flow run.
+
+        Ensures that necessary values are present in the job body and that the
+        job body is valid.
+
+        Args:
+            flow_run: The flow run to prepare the job configuration for
+            deployment: The deployment associated with the flow run used for
+                preparation.
+            flow: The flow associated with the flow run used for preparation.
+        """
+        super().prepare_for_flow_run(flow_run, deployment, flow)
+        self._populate_envs()
+        self._populate_image_if_not_present()
+        self._populate_command_if_not_present()
+        self._populate_name_if_not_present(flow_run)
+
+    def _populate_envs(self):
+        """Populate environment variables. BaseWorker.prepare_for_flow_run handles
+        putting the environment variables in the `env` attribute. This method
+        moves them into the jobs body"""
+        envs = [{"name": k, "value": v} for k, v in self.env.items()]
+        self.job_body["spec"]["template"]["spec"]["template"]["spec"]["containers"][0][
+            "env"
+        ] = envs
+
+    def _populate_name_if_not_present(self, flow_run):
+        """Adds the flow run name to the job if one is not already provided."""
+        try:
+            if "name" not in self.job_body["metadata"]:
+                self.job_body["metadata"]["name"] = flow_run.name
+        except KeyError:
+            raise ValueError("Unable to verify name due to invalid job body template.")
+
+    def _populate_image_if_not_present(self):
+        """Adds the latest prefect image to the job if one is not already provided."""
+        try:
+            if (
+                "image"
+                not in self.job_body["spec"]["template"]["spec"]["template"]["spec"][
+                    "containers"
+                ][0]
+            ):
+                self.job_body["spec"]["template"]["spec"]["template"]["spec"][
+                    "containers"
+                ][0]["image"] = f"docker.io/{get_prefect_image_name()}"
```

Comment:
running from my laptop gives `docker.io/prefecthq/prefect:dev-python3.10` which doesn't work on CloudRun, I'm not sure if we should force this to `docker.io/prefecthq/prefect:2-latest`?